example/lib/peripheral_details/components/property_row.dart

```diff
@@ -0,0 +1,170 @@
+import 'package:blemulator_example/styles/custom_text_style.dart';
+import 'package:flutter/material.dart';
+
+enum PropertyRowAccessoryPosition {
+  titleRow,
+  valueRow,
+  full,
+}
+
+class PropertyRow extends StatelessWidget {
+  final IconData titleIcon;
+  final String title;
+  final Color titleColor;
+  final String value;
+  final String valueCompanion;
+  final Widget accessory;
+  final PropertyRowAccessoryPosition accessoryPosition;
+
+  PropertyRow({
+    this.titleIcon,
+    this.title,
+    this.titleColor,
+    this.value = '',
+    this.valueCompanion,
+    this.accessory,
+    this.accessoryPosition = PropertyRowAccessoryPosition.full,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Card(
+      margin: EdgeInsets.all(8.0),
+      child: _buildCardBody(),
+    );
+  }
+
+  Widget _buildCardBody() {
+    switch (accessoryPosition) {
+      case PropertyRowAccessoryPosition.titleRow:
+      case PropertyRowAccessoryPosition.valueRow:
+        {
+          return _buildMainColumn();
+        }
+      case PropertyRowAccessoryPosition.full:
+      default:
+        {
+          return Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: <Widget>[
+              _buildMainColumn(),
+              _buildAccessory(),
```

Comment:
What is accessory? Why is there a value `full` if it isn't handled in a special way?

Reply:
Accessory is a widget that serves as an 'accessory' to the whole PropertyRow widget. You can place something like switch, button, disclosure indicator, icon or anything you like basically. Name `accessory` was used, because it was inspired by the similar view that exists in iOS' UITableViewCell.

`full` is handled just like default (there is no break in `full`s case so it falls through to the `default` case). This notation (with `default` case)  was used, because otherwise there was a warning, saying that `_buildCardBody` does not return `Widget` in every possible case (so it assumed that switch was not exhaustive, even though all 3 cases of `PropertyRowAccessoryPosition` were handled).

Reply:
Eventually, I've refactored switch statements with `PropertyRowAccessoryPosition`, because `default` case was only added courtesy of Flutter analyzer's warning even though all possible cases were handled (reference: https://github.com/dart-lang/sdk/issues/37498).

Implementation of `default` case returns `null` now, because it should not occur at all in the application.